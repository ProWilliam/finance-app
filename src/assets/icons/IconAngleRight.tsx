import * as React from "react";
import Svg, { SvgProps, Path } from "react-native-svg";

const IconAngleRight = (props: SvgProps) => (
  <Svg
    xmlns="http://www.w3.org/2000/svg"
    width={25}
    height={25}
    fill="none"
    aria-hidden="true"
    {...props}
    viewBox="0 0 24 24"
  >
    <Path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      d="m9 5 7 7-7 7"
    />
  </Svg>
)
export default IconAngleRight;