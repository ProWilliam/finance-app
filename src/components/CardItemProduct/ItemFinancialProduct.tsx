import React, { useState } from 'react';
import { TouchableOpacity, Text, View } from 'react-native';



// Hook and Component
import useAppNavigation from '../../hooks/useAppNavigation';
import IconAngleRight from '../../assets/icons/IconAngleRight';

// Styles
import style from './ItemFinancialProduct.styles';
import ItemFinancialProductProps from '../../types/components/ItemFinancialProduct.types';

const ItemFinancialProduct: React.FC<ItemFinancialProductProps> = ({ id, name }) => {

  const [route, setRoute] = useState('');
  const { appNavigation } = useAppNavigation();

  appNavigation(route, {id});

  return (
    <TouchableOpacity 
      style={style.containerItem}
      onPress={() => {setRoute('info')}}   
    >
      <View>
        <Text style={style.text}>{name}</Text>
        <Text style={style.textId}>ID:{id}</Text>
      </View>
      <IconAngleRight style={style.icon} />
    </TouchableOpacity>
  );
}

export default ItemFinancialProduct;