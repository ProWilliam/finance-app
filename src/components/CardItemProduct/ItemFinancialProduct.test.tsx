import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import ItemFinancialProduct from './ItemFinancialProduct';

import config from '../../../app.config';
import styles from './ItemFinancialProduct.styles';
import IconAngleRight from '../../assets/icons/IconAngleRight';

// Mock useAppNavigation hook
const mockAppNavigation = jest.fn();

jest.mock('../../hooks/useAppNavigation', () => ({
  __esModule: true,
  default: () => ({
    appNavigation: mockAppNavigation,
  }),
}));

describe('<ItemFinancialProduct />', () => {

  it('renders correctly with given props', () => {
    const component = renderer.create(<ItemFinancialProduct id="123" name="Product Name" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('updates route state and triggers navigation on press', () => {
    const component = renderer.create(<ItemFinancialProduct id="1" name="Product Name 1" />);

    act(() => {
      component.root.findByType(TouchableOpacity).props.onPress();
    });
  
    expect(mockAppNavigation).toHaveBeenCalledWith(config.extra.rootInfo, { id: '1' });
  });

  it('displays the correct id and name', () => {
    const testRenderer = renderer.create(<ItemFinancialProduct id="1" name="Product Name 1" />);
    const testInstance = testRenderer.root;
  
    const textName = testInstance.findByProps({ style: styles.text }).props.children;
    const texts =  testInstance.findAllByType(Text);
    const textID = texts.map(text => text.props.children)[1].join('');
  
    expect(textName).toBe('Product Name 1');
    expect(textID).toBe('ID:1');
  });

  it('renders the right icon', () => {
    const testRenderer = renderer.create(<ItemFinancialProduct id="1" name="Product Name 1" />);
    const testInstance = testRenderer.root;
  
    const icon = testInstance.findByType(IconAngleRight);
    expect(icon).toBeTruthy();
  });
  
});