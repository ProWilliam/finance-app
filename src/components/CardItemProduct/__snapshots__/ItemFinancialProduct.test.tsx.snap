// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemFinancialProduct /> renders correctly with given props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "padding": 9,
    }
  }
>
  <View>
    <Text
      style={
        {
          "backgroundColor": "#FFFFFF",
          "fontSize": 14,
          "marginBottom": 2,
        }
      }
    >
      Product Name
    </Text>
    <Text
      style={
        {
          "color": "#6a6a6a",
          "fontSize": 14,
        }
      }
    >
      ID:
      123
    </Text>
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    aria-hidden="true"
    bbHeight={25}
    bbWidth={25}
    fill="none"
    focusable={false}
    height={25}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "color": "#eaeaea",
          "fontSize": 35,
          "marginTop": 5,
        },
        {
          "flex": 0,
          "height": 25,
          "width": 25,
        },
      ]
    }
    tintColor="#eaeaea"
    vbHeight={24}
    vbWidth={24}
    width={25}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      font={
        {
          "fontSize": 35,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="m9 5 7 7-7 7"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
